cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaClusterRebalanceAssemblyOperator.java

```diff
@@ -391,22 +404,23 @@ public void onClose(KubernetesClientException e) {
                                     if (userTaskResult.succeeded()) {
                                         CruiseControlResponse response = userTaskResult.result();
                                         JsonObject taskStatusJson = response.getJson();
-                                        String taskStatus = taskStatusJson.getString("Status");
+                                        String taskStatusStr = taskStatusJson.getString("Status");
+                                        CruiseControlUserTaskStatus taskStatus = CruiseControlUserTaskStatus.lookup(taskStatusStr);
                                         switch (taskStatus) {
-                                            case CRUISE_CONTROL_TASK_COMPLETED:
+                                            case COMPLETED:
                                                 vertx.cancelTimer(t);
                                                 p.complete(new KafkaClusterRebalanceStatusBuilder()
                                                         .withSessionId(null)
                                                         .withOptimizationResult(taskStatusJson.getJsonObject("rebalance").getMap())
                                                         .addNewCondition().withType(State.ProposalReady.toString()).endCondition().build());
                                                 break;
-                                            case CRUISE_CONTROL_TASK_IN_PROGRESS:
-                                            case CRUISE_CONTROL_TASK_PENDING:
-                                            case CRUISE_CONTROL_TASK_ABORTING:
-                                                // just continue to do the periodic user task status request
+                                            case COMPLETED_WITH_ERROR:
+                                                // TODO: Add exception handling and update status?
+                                                break;
+                                            case IN_EXECUTION: // Skip as still processing
+                                                break;
+                                            case ACTIVE: // Skip as still processing
                                                 break;
```

Comment:
I was stupid I think, I didn't see this `break` here? it's enough for all the above case instead of having a `break` for each of them. The fall through will go to the last break.

Reply:
You aren't stupid, I just pushed a commit to add them ;-) (to fix my stupidity). I confess I am not that familiar with the behavior of Java `switch case` statements. I like being explicit where possible.